processor/src/trace/utils.rs

```diff
@@ -190,10 +191,64 @@ impl ChipletsLengths {
     }
 }
 
+// AUXILIARY COLUMN BUILDER
+// ================================================================================================
+
+/// Defines a builder responsible for building a single column in an auxiliary segment of the
+/// execution trace.
+pub trait AuxColumnBuilder<E: FieldElement<BaseField = Felt>> {
+    // REQUIRED METHODS
+    // --------------------------------------------------------------------------------------------
+
+    fn get_requests_at(&self, main_trace: &MainTrace, alphas: &[E], row_idx: usize) -> E;
+
+    fn get_responses_at(&self, main_trace: &MainTrace, alphas: &[E], row_idx: usize) -> E;
+
+    // PROVIDED METHODS
+    // --------------------------------------------------------------------------------------------
+
+    fn init_requests(&self, _main_trace: &MainTrace, _alphas: &[E]) -> E {
+        E::ONE
+    }
+
+    fn init_responses(&self, _main_trace: &MainTrace, _alphas: &[E]) -> E {
+        E::ONE
+    }
+
+    /// Builds the chiplets bus auxiliary trace column.
+    fn build_aux_column(&self, main_trace: &ColMatrix<Felt>, alphas: &[E]) -> Vec<E> {
+        let main_trace = MainTrace::new(main_trace);
+        let mut responses_prod: Vec<E> = unsafe { uninit_vector(main_trace.num_rows()) };
+        let mut requests: Vec<E> = unsafe { uninit_vector(main_trace.num_rows()) };
+
+        responses_prod[0] = self.init_responses(&main_trace, alphas);
+        requests[0] = self.init_requests(&main_trace, alphas);
+
+        let mut requests_running_prod = E::ONE;
+        for row_idx in 0..main_trace.num_rows() - 1 {
+            responses_prod[row_idx + 1] =
+                responses_prod[row_idx] * self.get_responses_at(&main_trace, alphas, row_idx);
+            requests[row_idx + 1] = self.get_requests_at(&main_trace, alphas, row_idx);
+            requests_running_prod *= requests[row_idx + 1];
+        }
+
+        let mut requests_running_divisor = requests_running_prod.inv();
+        let mut result_aux_column = responses_prod;
+        for i in (0..main_trace.num_rows()).rev() {
+            result_aux_column[i] *= requests_running_divisor;
+            requests_running_divisor *= requests[i];
+        }
+        result_aux_column
+    }
+}
+
+use miden_air::trace::main_trace::MainTrace;
+use vm_core::FieldElement;
```

Comment:
I would move these two imports and also the import on line 251 to the top of the file.

Reply:
done